### PR TITLE
Update uv mapping of cylinder caps to match CircleGeometry

### PR DESCRIPTION
**Pull request by abelnation**
This PR updates the uv mapping for CylinderGeometry and CylinderBufferGeometry to match the behavior of CircleGeometry (end cap acts as a circular cookie cutter on your textrure, instead of warping it around the center).  

Addresses #8270

### CODE REVIEW

**Comment by WestLangley** [comment]
@abelnation Thanks. BTW, since you are interested in formatting issues, paste this PR into http://zz85.github.io/mrdoobapproves/ to see the style differences.

/ping @Mugen87 for opinion.

File: src/extras/geometries/CylinderBufferGeometry.js [hunk 1]
```diff
@@ -54,7 +56,10 @@ THREE.CylinderBufferGeometry = function( radiusTop, radiusBottom, height, radial
 
 	// helper variables
 
-	var index = 0, indexOffset = 0, indexArray = [], halfHeight = height / 2;
+	var index = 0,
+		indexOffset = 0,
```
Comment (Mugen87) [inline src/extras/geometries/CylinderBufferGeometry.js]: This formatting here looks a little bit displaced.

File: src/extras/geometries/CylinderBufferGeometry.js [hunk 2]
```diff
@@ -208,7 +214,9 @@ THREE.CylinderBufferGeometry = function( radiusTop, radiusBottom, height, radial
 
 	function generateCap( top ) {
 
-		var x, centerIndexStart, centerIndexEnd;
+		var x,
+			centerIndexStart,
```
Comment (Mugen87) [inline src/extras/geometries/CylinderBufferGeometry.js]: Same here...

**Comment by abelnation** [comment]
@Mugen87 github displays tabs as 8 spaces, when most people use 4 (and 4 looks fine) in their editor.  This is one of (IMO) many problems with uses tabs instead of spaces for your code :P

**Comment by abelnation** [comment]
though, in looking at mrdoobapproves, it looks like his formatter handles multi-var decl's explicitly with 4 spaces instead of a tab.  will fix

File: src/extras/geometries/CylinderBufferGeometry.js [hunk 3]
```diff
@@ -227,23 +235,14 @@ THREE.CylinderBufferGeometry = function( radiusTop, radiusBottom, height, radial
 		for ( x = 1; x <= radialSegments; x ++ ) {
 
 			// vertex
-			vertices.setXYZ( index, 0, halfHeight * sign, 0 );
+			vertices.setXYZ( index, 0.5, halfHeight * sign, 0.5 );
```
Comment (Mugen87) [inline src/extras/geometries/CylinderBufferGeometry.js]: Why 0.5 for x and z?

Comment (abelnation) [inline src/extras/geometries/CylinderBufferGeometry.js]: great question.  i think i got mixed up w/ the uv values.  corrected this and verified that the result looks good

**Comment by mrdoob** [comment]
Hmm, could you not include the builds in the PR?

**Comment by mrdoob** [comment]
I wonder if there is a way on git of disallowing people to include the `builds` folder in their PRs...

**Comment by abelnation** [comment]
doh! sorry, accidentally committed those.  reverted

**Comment by abelnation** [comment]
could you add those files to the `.gitignore` making it such that you have to explicitly commit the files to update them in the repo?  not sure if gitignore's work that way or not

**Comment by mrdoob** [comment]
Thanks!

File: src/extras/geometries/CylinderBufferGeometry.js [hunk 4]
```diff
@@ -20,6 +20,7 @@ THREE.CylinderBufferGeometry = function( radiusTop, radiusBottom, height, radial
 	};
 
 	var scope = this;
+	var twoPi = 2.0 * Math.PI;
```
Comment (mrdoob) [inline src/extras/geometries/CylinderBufferGeometry.js]: Wait, this new variable is used only once?

**Comment by tschw** [comment]
@mrdoob 
> I wonder if there is a way on git of disallowing people to include the builds folder in their PRs...

We could invoke
```shell
git update-index --skip-worktree ./build/three.js
git update-index --skip-worktree ./build/three.min.js
```
in the build script.

This way the local git index of the user would be configured to always ignore builds as if they were in `.gitignore`, but (other than with `.gitignore`) they can still be part of the repository. You just have to think of an exception for yourself, so you can still update them. A customized build script only on your disk and listed in `.gitignore` might do the trick. An environment variable set in your shell profile that suppresses this behavior would be another simple option.

@abelnation
> though, in looking at mrdoobapproves, it looks like his formatter handles multi-var decl's explicitly with 4 spaces instead of a tab. will fix

Is this actually something that needs fixing? Or more of a quirk of mrdoobapproves? 

I have no problem typing Mr.Doob's Code Type (TM) by hand - I love it. However, once there are per-case rules for tab expansion my tools stop working smoothly. This fix reads terribly misguided to me. Also, I don't remember reading any of this in the style guide.

> reverting build files to dev

Something on the pro-side of squashed commits: Stuff like this is much less of a mess than before, except a workable history of the feature branch is actually needed, of course.

Seeing many commit messages with the same name, I guess you may still want to [tidy up](https://github.com/mrdoob/three.js/wiki/A-Contributor's-Guide-to-the-Git-Galaxy).

**Comment by tschw** [comment]
Two ideas regarding this submission:

- Does it tolerate manual horizontal alignment? So e.g. matrices formatted as matrices won't be destroyed?
- How about parsing `git diff` output and format edited lines only?
- Maybe we should reject code that contains certain white space errors by git configuration?

File: src/extras/geometries/CylinderBufferGeometry.js (hunk 4 above)
Comment (Mugen87) [inline src/extras/geometries/CylinderBufferGeometry.js]: Indeed! I don't think the variable is necessary here.

@abelnation Is this something you can handle? :wink:

Comment (abelnation) [inline src/extras/geometries/CylinderBufferGeometry.js]: done

**Comment by abelnation** [comment]
@tschw sorry, i'm not sure if you're suggesting/requesting any other action items from me in your previous comment?  would you like me to rebase things in this PR or are you just giving suggestions/thoughts for next time?  

Happy to make any follow-up changes, just let me know.

**Comment by tschw** [comment]
@abelnation 
> sorry, i'm not sure if you're suggesting/requesting any other action items from me in your previous comment? would you like me to rebase things in this PR or are you just giving suggestions/thoughts for next time? 

In my first thread I was replying to @mrdoob 's thought about forcing builds to be ignored. I'm also giving some suggestions for an easier life with git: Since recently, all commits are being squashed, so keeping your history clean is pretty much for your own good only. I usually still try to give feature branches meaningful and well-structured commits because it makes them easier to read and also easier to change something after the merge (if it's not too long ago).

Further, I am ranting about the use of 4 spaces for tabs. Reading your post again, I might have gotten that wrong and you meant "fix" = "use a tab instead of the four spaces" and that's all fine. I really think we should be using one or the other - not both - and since we usually use tabs there should be no 4 spaces. Inconsistencies of that kind are awkward when using standard tools and doing the formatting manually.

In my second post, I think I mixed this thread up with one about automatic formatting :). Please excuse the noise!

**Comment by abelnation** [comment]
@tschw great, that was what i thought you might have meant.  

i'm a bit of a n00b with regard to rebasing (especially when rebasing commits that have been pushed to origin), but i'll brush up on my git skillz for future PR's.

**Pull request merged**